lms/djangoapps/certificates/views/webview.py

```diff
@@ -517,6 +517,13 @@ def render_html_view(request, user_id, course_id):
         log.info(error_str, course_id, user_id, str(exception))
         return _render_invalid_certificate(course_id, platform_name, configuration)
 
+    if not certificates_viewable_for_course(course):
+        log.info(
+            "Invalid cert: Certificate for %s is not viewable yet.",
+            course_id,
+        )
+        return _render_invalid_certificate(course_id, platform_name, configuration)
+
     # Kick the user back to the "Invalid" screen if the feature is disabled for the course
     if not course.cert_html_view_enabled:
```

Comment:
@ahsan-ul-haq You can combine both these condition with `or` and return `_render_invalid_certificate` if any condition fails.